### PR TITLE
sys-apps/dstat: upgrade to python 3.12

### PR DESCRIPTION
**Pull request by lmiphay**
add python 3.12
fix use of unescaped backslashes in regex strings

Bug: https://bugs.gentoo.org/929807
Signed-off-by: Paul Healy <lmiphay@gmail.com>

### CODE REVIEW

File: sys-apps/dstat/dstat-0.7.4-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (arthurzam) [inline sys-apps/dstat/dstat-0.7.4-r3.ebuild:4]: can you bump to EAPI=8?

File: sys-apps/dstat/dstat-0.7.4-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+PYTHON_COMPAT=( python3_{9..12} )
```
Comment (arthurzam) [inline sys-apps/dstat/dstat-0.7.4-r3.ebuild:6]:
```suggestion
PYTHON_COMPAT=( python3_{10..12} )
```

File: sys-apps/dstat/dstat-0.7.4-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		dodoc examples/{mstat,read}.py
+	fi
+	if use doc; then
+		dodoc docs/*.html
+	fi
+}
+
+src_test() {
+	python_foreach_impl emake test
+}
```
Comment (arthurzam) [inline sys-apps/dstat/dstat-0.7.4-r3.ebuild:61]: put it before `src_install`, since it is called before install

**Review by arthurzam — CHANGES_REQUESTED** [review]

**Comment by lmiphay** [comment]
Updated as per comments. Thanks!

**Pull request closed without merging**